### PR TITLE
[deutschebahn] DB API Marketplace adjustments

### LINKED ISSUE #12785: [deutschebahn] Update to DB API Marketplace
The DB Open-Data-Portal will be replaced by the DB API Marketplace (see [FAQ Post](https://data.deutschebahn.com/faq-api-migration)), so the binding has to be adjusted to work with the new api.

The URL for the api has changed and the authentication now requires two headers instead of one.

### PR DESCRIPTION
**Pull request by soenkekueper**
The URL and the authentication Headers have changed, due the DB will move the timetables api from the Open-Data-Portal to the  DB API Marketplace middle this year.

Closes #12785

### CODE REVIEW

File: bundles/org.openhab.binding.deutschebahn/src/main/resources/OH-INF/i18n/deutschebahn_de.properties [hunk 1]
```diff
@@ -18,12 +18,14 @@ thing-type.deutschebahn.train.group.trip.description = Enthält alle Information
 
 # thing types config
 
-thing-type.config.deutschebahn.timetable.accessToken.label = Zugriffsschlüssel
-thing-type.config.deutschebahn.timetable.accessToken.description = Zugriffsschlüssel für die Timetable V1 API aus dem Developer-Portal der DeutschenBahn.
+thing-type.config.deutschebahn.timetable.clientId.label = Client Id
```
Comment (jlaur) [inline bundles/org.openhab.binding.deutschebahn/src/main/resources/OH-INF/i18n/deutschebahn_de.properties:21]: Translations should be provided through Crowdin. Please see https://github.com/openhab/openhab-addons/blob/main/README.md#translations.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks, LGTM. Only a minor request to remove the German translation update to avoid Crowdin conflicts.

File: bundles/org.openhab.binding.deutschebahn/src/main/resources/OH-INF/i18n/deutschebahn_de.properties (hunk 1 above)
Comment (soenkekueper) [reply bundles/org.openhab.binding.deutschebahn/src/main/resources/OH-INF/i18n/deutschebahn_de.properties]: Thanks for your review.
I've reverted the changes and applied them in crowdin.

**Review by jlaur — APPROVED** [review]
LGTM - thanks for the adjustments!

**Pull request merged**